smarts/core/vehicle_index.py

```diff
@@ -372,7 +342,17 @@ def build_agent_vehicle(
             initial_speed,
         )
 
-        waypoint_paths = sim.waypoints.waypoint_paths_at(vehicle.position, lookahead=1)
+        if vehicle_exists:
+            old_vehicle = self._vehicles[vehicle_id]
```

Comment:
This code is a much lower level of abstraction, we should aim to keep the same abstraction levels for each line within a method. IOW, we should move this out into a sep. method.